test/plugins/tokens.test.js

```diff
@@ -218,24 +225,119 @@ describe('token plugin test', () => {
         });
     });
 
-    describe('DELETE /tokens/{id}', () => {
+    describe('PUT /tokens/{id}', () => {
         let options;
-        const tokenId = testToken.id;
 
         beforeEach(() => {
             options = {
-                method: 'DELETE',
+                method: 'PUT',
                 url: `/tokens/${tokenId}`,
                 credentials: {
                     username,
                     scope: ['user']
+                },
+                payload: {
+                    name: testToken.name,
+                    description: 'a new description'
                 }
             };
 
             tokenFactoryMock.get.resolves(tokenMock);
+            userMock.tokens = tokensGetterMock([tokenMock]);
+            userFactoryMock.get.withArgs({ username }).resolves(userMock);
+        });
+
+        it('returns 404 when the token does not exist', () => {
+            tokenFactoryMock.get.resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 404 when the user does not exist', () => {
+            userFactoryMock.get.withArgs({ username }).resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 409 when a token with the same name already exists', () => {
+            options.url = `/tokens/${tokenId + 1}`;
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 409);
+                assert.strictEqual(reply.result.message,
+                    `Token with name ${testToken.name} already exists`);
+            });
+        });
+
+        it('returns 200 if update successfully', () => {
+            const expected = hoek.applyToDefaults(testToken, {
+                description: 'a new description'
+            });
+
+            tokenMock.toJson.returns(expected);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 200);
+                assert.calledOnce(tokenMock.update);
+                assert.deepEqual(reply.result, expected);
+            });
+        });
+
+        it('doesn\'t error if the new name conflicts only with the old name', () => {
```

Comment:
From a testing perspective, this looks no different than this test https://github.com/screwdriver-cd/screwdriver/pull/573/files#diff-ee5dae69c9de402833a459d19c780d18R276